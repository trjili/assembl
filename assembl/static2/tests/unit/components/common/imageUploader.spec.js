import React from 'react';
import renderer from 'react-test-renderer';

import ImageUploader from '../../../../js/app/components/common/imageUploader';

describe('ImageUploader component', () => {
  it('should render an image uploader', () => {
    const handleChangeSpy = jest.fn(() => {});
    const file = new File([''], 'foobar.png');
    const component = renderer.create(<ImageUploader file={file} handleChange={handleChangeSpy} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});