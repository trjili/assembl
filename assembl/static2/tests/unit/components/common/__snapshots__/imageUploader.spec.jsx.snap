exports[`ImageUploader component should render an image uploader 1`] = `
<div>
  <div>
    <span>
      foobar.png
    </span>
    <img
      alt="foobar.png"
      className="img-thumbnail"
      src="" />
  </div>
  <button
    className="btn btn-default"
    disabled={false}
    onClick={[Function]}
    type="button">
    <span>
      Upload Button
    </span>
  </button>
  <input
    accept="image/*"
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file" />
</div>
`;
